mne/decoding/xdawn.py

```diff
@@ -0,0 +1,182 @@
+    ----------
+    n_components : int (default 2)
+        The number of components to decompose M/EEG signals.
+    signal_cov : None | Covariance | ndarray, shape (n_channels, n_channels)
+        (default None). The signal covariance used for whitening of the data.
+        if None, the covariance is estimated from the epochs signal.
+    reg : float | str | None (default None)
+        if not None, allow regularization for covariance estimation
+        if float, shrinkage covariance is used (0 <= shrinkage <= 1).
+        if str, optimal shrinkage using Ledoit-Wolf Shrinkage ('ledoit_wolf')
+        or Oracle Approximating Shrinkage ('oas').
+
+    Attributes
+    ----------
+    filters_ : dict of ndarray
+        If fit, the Xdawn components used to decompose the data for each event
+        type, else empty.
+    patterns_ : dict of ndarray
+        If fit, the Xdawn patterns used to restore M/EEG signals for each event
+        type, else empty.
+
+
+    See Also
+    --------
+    mne.preprocessing.Xdawn
+    CSP
+
+    References
+    ----------
+    [1] Rivet, B., Souloumiac, A., Attina, V., & Gibert, G. (2009). xDAWN
+    algorithm to enhance evoked potentials: application to brain-computer
+    interface. Biomedical Engineering, IEEE Transactions on, 56(8), 2035-2043.
+
+    [2] Rivet, B., Cecotti, H., Souloumiac, A., Maby, E., & Mattout, J. (2011,
+    August). Theoretical analysis of xDAWN algorithm: application to an
+    efficient sensor selection in a P300 BCI. In Signal Processing Conference,
+    2011 19th European (pp. 1382-1386). IEEE.
+    """
+
+    def __init__(self, n_components=2, signal_cov=None, reg=None):
+        """init xdawn."""
+        # correct_overlap is not yet supported
+        # This attribute is introduced to make it compatible with parent
+        # class
+        self.correct_overlap = False
+        super(XdawnTransformer, self).__init__(n_components, signal_cov, reg)
+
+    def fit(self, X, y):
+        """Fit Xdawn from epochs.
+
+        Parameters
+        ----------
+        X : ndarray, shape(n_channels, n_times, n_freq)
+            Data of epochs.
+        y : ndarray shape(n_samples,)
+            Target values.
+
+        Returns
+        -------
+        self : XdawnTransformer instance
+            The XdawnTransformer instance.
+        """
+
+        if X.ndim != 3 or not isinstance(X, np.ndarray):
+            raise ValueError("X should be 3 dimensional ndarray")
+        if not isinstance(y, np.ndarray):
+            raise ValueError("Labels must be numpy array")
+
+        self._n_chan = len(X[1])
+        event_id = dict()
+        for idx in y:
+            event_id['cond' + str(idx)] = idx
+
+        # Extract signal covariance
+        self._get_signal_cov(X)
+
+        # estimates evoked covariance
+        self._fit_xdawn(X, y, event_id)
+        self.event_id = event_id
+        self.epochs_data = X
+        self.exclude = list(range(self.n_components, self._n_chan))
```

Comment:
@dengemann same question here: do we break the api and correct the attribute name or leave it for now?